src/pages/Background/index.ts

```diff
@@ -12,6 +12,7 @@ import OnClickData = chrome.contextMenus.OnClickData;
 import OnInputEnteredDisposition = chrome.omnibox.OnInputEnteredDisposition;
 
 const browser = getBrowser();
+let configuration = null; // this js needs init function which initializes the configuration variable thus solving many lines in this file which repeadiatly uses getConfig() it might be better 
```

Comment:
i come from the java/kotlin/python so I don't know more of vite/javascript/typescript but this variable needs to be initialized in init function which should be added to this file which also reduces the getConfig() function code.